src/realm/sync/subscriptions.cpp

```diff
@@ -749,25 +671,146 @@ SubscriptionSet SubscriptionStore::get_by_version(int64_t version_id) const
     return get_by_version_impl(version_id, util::none);
 }
 
+void SubscriptionStore::update_live_sub_sets(const SubscriptionSet& new_set)
+{
+    std::lock_guard<std::mutex> lk(m_live_sub_sets_mutex);
+    if (new_set.state() == SubscriptionSet::State::Complete &&
+        new_set.version() > m_active_subscription_set.version()) {
+        m_active_subscription_set = new_set;
+        m_active_version = m_active_subscription_set.version();
+    }
+    else if (new_set.version() >= m_latest_subscription_set.version()) {
```

Comment:
use m_latest_version, or get rid of it altogether. Similar for m_active_version.